### ISSUE TITLE
Pathpanner Probleme

### ISSUE THREAD

**Issue by marvin78** [EinEinfach/CaSSAndRA#38]
Ich schreibe das hier nun mal in Deutsch, da mir alles andere zu anstrengend ist. Der Pathplanner hat doch noch so einige Probleme. Insbesondere auf bestimmten Karten mit Tasks. Kartenweiterungen führen zu unerwarteten Problemen. Ich zähle hier mal ein paar Punkte auf, die ergänzt werden können:

1. Bekannt aus dem Ardumower Forum: Der Pathplanner nimmt seltsame Wege. Insbesondere, wenn Tasks auf einer Karte kombiniert werden, fährt er oft Umwege über den Perimeter und das sehr oft andersherum, als es "erlaubt" wäre. Dabei wird nicht die Einstellung beachtet, die man dem Task/den Tasks mitgegeben hat. Das führt zu einer deutlich längeren Bearbeitungszeit der Tasks, als es nötig wäre und auch zu Kollisionen die nicht nötig wären. Lösungsvorschläge sind: den Parameter nie falsch herum abfahren, Referenzpunkte manuell setzbar machen, sodass der Mäher immer, wie gewünscht in einen Task herein fährt. Das wäre für viele Dinge optimal, da man sehr viel Einfluss nehmen könnte und Zeit und Energie gespart würde. 

3. Ich habe eine Karte (1), die einen Teil einer anderen Karte (2) darstellt. Die Pfadplanung auf der Karte 1 ist ok und der Mäher fährt wenige Umwege, er stößt selten an, da er den Pfad weitgehend so abfährt, wie es bei der Planung der Karte gedacht war. Auf Karte 2 wird absolut Chaotisch gefahren. Das seltsame ist, das zwar schon die sichtbaren Pfadlinien chaotischer sind, als auf Karte 1 aber die Realität sieht noch schlimmer aus. Er fährt teilweise Kreuz und quer über die Karte, wie ein Chaosmäher. Dabei ist seltsam, dass bestimmte Wege, die er wählt, so gar nicht eingezeichnet sind. Trotzdem "hüpft" der Mäher von Perimetergenze zu gegenüberliegender Grenze. Auf Karte 1 mit dem gleichen Task, macht er das nicht. 

4. Die Pfadplanung mit Tasks auf Teilausschnitten scheint komplett unterschiedlich zu sein zur Planung auf einer Gesamtkarte, die identisch zu den Taskgrenzen auf einer anderen Karte ist.

5. Folgendes passiert auf einer bestimmten Karte (2): Hier möchte der Mäher, um den markierten Punkt zu erreichen, mitten durch den Pool. Es hat nur deshalb nicht platsch und blubb gemacht, weil der Pool eine Schiebehalle hat, die geschlossen war. Das ist reproduzierbar mit diesem Task. 
Den gleichen Task habe ich auf einer Karte, die identisch ist, bis auf den Anschluss der Fläche oben links auf dem Screenshot. Dort passiert das nicht. Auch passiert es nicht, wenn ich vor diesem Tasks einen weiteren hinzufüge. Dann arbeitet er die Tasks ab und nimmt übliche Wege. 
![image](https://github.com/EinEinfach/CaSSAndRA/assets/10920356/17fcfd51-1537-4c14-82c6-5b17b0ceced8)
(Kartenaussschnitt)

**Comment by EinEinfach** [comment]
Hast du ENABLE_PASS = 1 in config.h?

**Comment by marvin78** [comment]
Nein. Trotzdem interessiert mich, was es damit zu tun haben könnte.

**Comment by EinEinfach** [comment]
siehe hier
[https://github.com/EinEinfach/CaSSAndRA/issues/35](url)

**Comment by marvin78** [comment]
Naja. Wie gesagt, die Encryption ist aus. Daran wird es also nicht liegen. Auch würde es nicht erklären, warum es nur bei der erweiterten Karte Probleme macht, nicht aber in der um eine Fläche kleine Karte, die sonst identisch ist. Und die anderen Punkte erklärt es natürlich auch nicht.

**Comment by EinEinfach** [comment]
Mir ging es nur um den Punkt, wo der Mäher fast im Pool gelandet ist. Das ist prio 1, da bin ich noch am tüfteln. Der Rest sind Pathplanner Themen

**Comment by disaster123** [comment]
Kann das nicht eventuell auch hiermit zu tun haben?

https://github.com/EinEinfach/CaSSAndRA/issues/33

**Comment by marvin78** [comment]
0.0 wäre ja Basis oder? Die ist in der anderen Richtung.
